CRM/Contact/BAO/Query.php

```diff
@@ -4734,17 +4734,26 @@ public static function appendAnyValueToSelect($selectClauses, $groupBy, $aggrega
    *  on full_group_by mode, then append the those missing columns to GROUP BY clause
    * keyword to select fields not present in groupBy
    *
-   * @param string $groupBy - GROUP BY clause where missing ORDER BY columns will be appended
+   * @param string|array $groupBy - GROUP BY clause or array of columns where missing ORDER BY columns will be appended if not present
    * @param array $orderBys - ORDER BY sub-clauses
    *
    */
   public static function getGroupByFromOrderBy(&$groupBy, $orderBys) {
     if (!CRM_Utils_SQL::disableFullGroupByMode()) {
       foreach ($orderBys as $orderBy) {
-        $orderBy = str_replace(array(' DESC', ' ASC', '`'), '', $orderBy); // remove sort syntax from ORDER BY clauses if present
+        $orderBy = str_ireplace(array(' DESC', ' ASC', '`'), '', $orderBy); // remove sort syntax from ORDER BY clauses if present
```

Comment:
so - this seems totally safe IMHO - if it's OK for non sensitive then case sensitive is ok